front/src/common/BootstrapSNCF/InputGroupSNCF.tsx

```diff
@@ -13,19 +13,18 @@ type Option = {
   unit?: string;
 };
 
-export type InputGroupSNCFValue = { type?: string; value?: string | number };
+export type InputGroupSNCFValue = { type?: string; unit: string; value?: string | number };
 
 type Props = {
   id: string;
   label?: React.ReactElement | string;
   options: Option[];
-  handleType: (type: InputGroupSNCFValue) => void;
+  unit: string;
```

Comment:
You can largely simplify the props, as you can remove:
- orientation
- placeholder
- sm
- typeValue
- condensed
- step
- textRight

- [ ] To simplify also the props, you could have a prop value: { value: number; unit: string }.
- [ ] You can also rename handleUnit => onChange, options => unitOptions, label => inputLabel